### ISSUE TITLE
Add a manual rootless mode

### ISSUE THREAD

**Issue by BtbN** [AdAway/AdAway#858]
My device is not rooted, but I do have a custom recovery installed.
AdAway is the only app that needs root I'd like to use.

There are multiple ways this could be done, the most simple one is just writing the new hosts file to /sdcard/, and telling the user where it was stored, so I can then boot into recovery and put it in the correct place via adb.

AdAway could also create an OpenRecovery script, like CM/LAOS does for their OTA updates, which then instructs TWRP to move the hosts file in place automatically, without any user interaction. AdAway might need to be installed as system App for this to work, not sure how exactly it works.

I'd like to avoid installing root, as that breaks several Apps I use regularly, and I don't really need it except for AdAway.

**Comment by GeoZac** [comment]
@BtbN you could make a host files on a rooted firmware and then push it via recovery's file manager.

**Comment by ipha** [comment]
What if /etc/hosts was a symlink to a /sdcard/* file? Assuming android follows the symlink it should allow rootless updates.

**Comment by mrplow** [comment]
I agree and I would also like to run adaway in the same manner as the submitter. Getting an installer zip would be ideal but for a now allowing adaway to start without root would be great.

To simplify things if no root is detected allow the app to start and only enable the option to save the hosts file to the internal storage.

My current work around is to download a hosts file online and to manually pull out my whitest entries but it's a pain on mobile.
